core/tests/puppeteer-acceptance-tests/functions/is-element-clickable.ts

```diff
@@ -0,0 +1,170 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Utility function to check if an element is clickable.
+ */
+
+export default function isElementClickable(element: Element): boolean {
+  /**
+   * This function gets the the overlapping element if any by checking the
+   * element that is present in the center of the target element's position.
+   */
+  const getOverlappingElement = (
+    element: Element,
+    parent: ShadowRoot | Document = document
+  ): Element | null => {
+    const elementDimensions = element.getBoundingClientRect();
+    const x = elementDimensions.left + element.clientWidth / 2;
+    const y = elementDimensions.top + element.clientHeight / 2;
+
+    return parent.elementFromPoint(x, y);
+  };
+
+  /**
+   * This function also gets the overlapping element if any at the center
+   * of the given element. However this uses the client rect, which
+   * is applicable in special cases where the text is multiline. This
+   * applies to special cases like the span.
+   */
+  const getOverlappingRect = (
+    element: Element,
+    parent: ShadowRoot | Document = document
+  ): Element | null => {
+    const rects = element.getClientRects();
+    const rect = rects[0];
+    const x = rect.left + rect.width / 2;
+    const y = rect.top + rect.height / 2;
+
+    return parent.elementFromPoint(x, y);
+  };
+
+  /**
+   * This function combines the overlapping element and overlapping rect
+   * to get the element or elements that are overlapping the given
+   * element.
+   */
+  const getOverlappingElements = (
+    element: Element,
+    parent: ShadowRoot | Document = document
+  ): Element[] => {
+    const overlappingElement = getOverlappingElement(element, parent);
+    const overlappingRect = getOverlappingRect(element, parent);
+    const overlappingElements: Element[] = [];
+    if (overlappingElement) {
+      overlappingElements.push(overlappingElement);
+    }
+    if (overlappingRect) {
+      overlappingElements.push(overlappingRect);
+    }
+    return overlappingElements;
+  };
+
+  /**
+   * This function is recursive and checks if the target element is the only
+   * overlapping element meaning there are no other elements blocking
+   * the target element.
+   *
+   * @param {Element} targetElement - The element to compare with the other
+   * overlapping elements.
+   * @param {Element[]} overlappingElements - The overlapping elements to compare
+   * against the target element.
+   */
+  const isOverlappingElementsMatching = (
```

Comment:
Kind of confused about the method name -- and still not clear to me what this does based on the docstring. Can you document the return value clearly and explain how it relates to the input args (conceptually)?

Based on the top part of the docstring I would have expected something like isNotOverlapped(targetElement) but then I don't know what overlappingElements is for.

Reply:
Done